tests: bluetooth.audio.ascs: build failure on all NXP platforms

### Describe the bug

build failure on all NXP platforms

the failure application is tests/bluetooth/audio/ascs

```
INFO    - To rerun the tests, call twister using the following commandline:
INFO    -
INFO    - west twister -p frdm_mcxa346/mcxa346 -s bluetooth.audio.ascs.test_stream_pair
INFO    - or with west:
INFO    - west build -p -b frdm_mcxa346/mcxa346 tests/bluetooth/audio/ascs -T bluetooth.audio.ascs.test_stream_pair
```

50a5cb3ec680b8021f1800ae2653849a0f794c66 is the first bad commit
commit 50a5cb3ec680b8021f1800ae2653849a0f794c66
Author: Emil Gydesen <emil.gydesen@nordicsemi.no>
Date:   Mon Sep 29 11:36:47 2025 +0200

    tests: Bluetooth: ASCS: Modify unittests to native_sim

    Modify the unit tests to use the native_sim board instead of the
    unit_testing board.
    The native_sim board allows us to use the kernel in a more
    regular way, while still being able to treat the tests as
    unittest.

    To support mocking we cannot rely on the default
    Kconfig options and values. The required Kconfig options
    to satisfy dependencies are overridden in the Kconfig
    file and the necessary Kconfig files are sourced.
    We cannot enable e.g. CONFIG_BT as that would attempt
    to pull in a lot of files we do not want in this test.

    This commit also removes unused includes in the CMakelists,
    and cleans up the code.
    Since the kernel is actually running, we also need to support
    that e.g. k_work items are actually scheduled, and that we have
    to wait for them to finish. test_drain_syswq has been implemented
    as a way to drain the system workqueue before we check for
    callbacks or perform subsequent actions.

    Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

 tests/bluetooth/audio/ascs/CMakeLists.txt          | 44 ++++++++++----
 tests/bluetooth/audio/ascs/Kconfig                 | 71 ++++++++++++++++++++++
 tests/bluetooth/audio/ascs/include/test_common.h   |  1 +
 tests/bluetooth/audio/ascs/prj.conf                | 13 +---
 tests/bluetooth/audio/ascs/src/main.c              | 14 +++++
 .../audio/ascs/src/test_ase_state_transition.c     | 54 ++++++++++++++++
 tests/bluetooth/audio/ascs/src/test_common.c       | 29 ++++++++-
 tests/bluetooth/audio/ascs/testcase.yaml           | 10 +--
 tests/bluetooth/audio/ascs/uut/CMakeLists.txt      | 29 ---------
 tests/bluetooth/audio/mocks/src/pacs.c             |  3 +-
 10 files changed, 208 insertions(+), 60 deletions(-)


### Regression

- [x] This is a regression.

### Steps to reproduce

west build -p -b frdm_mcxa346/mcxa346 tests/bluetooth/audio/ascs -T bluetooth.audio.ascs.test_stream_pair

### Relevant log output

```shell
In file included from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:7,
                 from /home/shared/disk/zephyr_project/zephyr_dev/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/common/fsl_common.h:12,
                 from /home/shared/disk/zephyr_project/zephyr_dev/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/port/fsl_port.h:11,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/soc/nxp/mcx/mcxa/./soc.h:12,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/modules/cmsis_6/./cmsis_core_m.h:24,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/modules/cmsis_6/./cmsis_core.h:10,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:24,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/include/zephyr/arch/arm/asm_inline.h:18,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/include/zephyr/arch/arm/arch.h:30,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/include/zephyr/arch/cpu.h:19,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/include/zephyr/kernel_includes.h:36,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/include/zephyr/kernel.h:17,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/include/zephyr/net_buf.h:16,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/include/zephyr/bluetooth/classic/classic.h:25,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/include/zephyr/bluetooth/bluetooth.h:38,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/include/zephyr/bluetooth/iso.h:30,
                 from /home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/ascs/src/main.c:18:
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: error: redefinition of 'struct mock_check_if_assert_expected_Fake'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: note: originally defined here
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: error: conflicting types for 'mock_check_if_assert_expected_Fake'; have 'struct mock_check_if_assert_expected_Fake'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: note: previous declaration of 'mock_check_if_assert_expected_Fake' with type 'mock_check_if_assert_expected_Fake'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: error: conflicting types for 'mock_check_if_assert_expected_fake'; have 'mock_check_if_assert_expected_Fake'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: note: previous declaration of 'mock_check_if_assert_expected_fake' with type 'mock_check_if_assert_expected_Fake'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/port/fsl_port.h: In function 'PORT_SetPinConfig':
/home/shared/disk/zephyr_project/zephyr_dev/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/port/fsl_port.h:362:5: error: implicit declaration of function 'assert' [-Werror=implicit-function-declaration]
  362 |     assert(config);
      |     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/port/fsl_port.h:12:1: note: 'assert' is defined in header '<assert.h>'; did you forget to '#include <assert.h>'?
   11 | #include "fsl_common.h"
  +++ |+#include <assert.h>
   12 |
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h: At top level:
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: error: redefinition of 'struct mock_check_if_assert_expected_Fake'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: note: originally defined here
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: error: conflicting types for 'mock_check_if_assert_expected_Fake'; have 'struct mock_check_if_assert_expected_Fake'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: note: previous declaration of 'mock_check_if_assert_expected_Fake' with type 'mock_check_if_assert_expected_Fake'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: error: conflicting types for 'mock_check_if_assert_expected_fake'; have 'mock_check_if_assert_expected_Fake'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:31: note: previous declaration of 'mock_check_if_assert_expected_fake' with type 'mock_check_if_assert_expected_Fake'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND'
 8632 | #define EXPAND(x) x
      |                   ^
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:37: note: in expansion of macro 'EXPAND'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                     ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:38: note: in expansion of macro 'EXPAND'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                      ^~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8689:45: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC0'
 8689 | #define DECLARE_FUNC_VALUE_N(N, ...) EXPAND(DECLARE_FAKE_VALUE_FUNC##N(__VA_ARGS__))
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8687:44: note: in expansion of macro 'DECLARE_FUNC_VALUE_N'
 8687 | #define DECLARE_FUNC_VALUE_(N, ...) EXPAND(DECLARE_FUNC_VALUE_N(N, __VA_ARGS__))
      |                                            ^~~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/subsys/testsuite/include/zephyr/fff.h:8685:16: note: in expansion of macro 'DECLARE_FUNC_VALUE_'
 8685 |         EXPAND(DECLARE_FUNC_VALUE_(PP_NARG_MINUS2(__VA_ARGS__), __VA_ARGS__))
      |                ^~~~~~~~~~~~~~~~~~~
/home/shared/disk/zephyr_project/zephyr_dev/zephyr/tests/bluetooth/audio/mocks/include/assert.h:14:1: note: in expansion of macro 'DECLARE_FAKE_VALUE_FUNC'
   14 | DECLARE_FAKE_VALUE_FUNC(bool, mock_check_if_assert_expected);
      | ^~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: ubuntu
- toolchain: Zephyr
- version: 

### Additional Context

_No response_

--- a/tests/bluetooth/audio/ascs/testcase.yaml
+++ b/tests/bluetooth/audio/ascs/testcase.yaml
@@ -2,13 +2,13 @@ common:
   tags:
     - bluetooth
     - bluetooth_audio
+  platform_allow:
+    - native_sim
+  integration_platforms:
+    - native_sim
 
 tests:
-  bluetooth.audio.ascs.test_default:
-    platform_allow:
-      - native_sim
-    integration_platforms:
-      - native_sim
+  bluetooth.audio.ascs.test_default: {}
   bluetooth.audio.ascs.test_snk_only:
     extra_configs:
       - CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT=0

--- a/tests/bluetooth/audio/pacs/testcase.yaml
+++ b/tests/bluetooth/audio/pacs/testcase.yaml
@@ -2,82 +2,55 @@ common:
   tags:
     - bluetooth
     - bluetooth_audio
+  platform_allow:
+    - native_sim
+  integration_platforms:
+    - native_sim
+
 tests:
   bluetooth.audio.pacs.test_default:
-    platform_allow:
-      - native_sim
-    integration_platforms:
-      - native_sim
     extra_configs:
       - CONFIG_BT_PAC_SNK=y
       - CONFIG_BT_PAC_SNK_LOC=y
       - CONFIG_BT_PAC_SRC=y
       - CONFIG_BT_PAC_SRC_LOC=y
   bluetooth.audio.pacs.test_snk_pac_snk_loc_src_pac:
-    platform_allow:
-      - native_sim
-    integration_platforms:
-      - native_sim
     extra_configs:
       - CONFIG_BT_PAC_SNK=y
       - CONFIG_BT_PAC_SNK_LOC=y
       - CONFIG_BT_PAC_SRC=y
       - CONFIG_BT_PAC_SRC_LOC=n
   bluetooth.audio.pacs.test_snk_pac_snk_loc:
-    platform_allow:
-      - native_sim
-    integration_platforms:
-      - native_sim
     extra_configs:
       - CONFIG_BT_PAC_SNK=y
       - CONFIG_BT_PAC_SNK_LOC=y
       - CONFIG_BT_PAC_SRC=n
       - CONFIG_BT_PAC_SRC_LOC=n
   bluetooth.audio.pacs.test_snk_pac:
-    platform_allow:
-      - native_sim
-    integration_platforms:
-      - native_sim
     extra_configs:
       - CONFIG_BT_PAC_SNK=y
       - CONFIG_BT_PAC_SNK_LOC=n
       - CONFIG_BT_PAC_SRC=n
       - CONFIG_BT_PAC_SRC_LOC=n
   bluetooth.audio.pacs.test_snk_pac_src_pac_src_loc:
-    platform_allow:
-      - native_sim
-    integration_platforms:
-      - native_sim
     extra_configs:
       - CONFIG_BT_PAC_SNK=y
       - CONFIG_BT_PAC_SNK_LOC=n
       - CONFIG_BT_PAC_SRC=y
       - CONFIG_BT_PAC_SRC_LOC=y
   bluetooth.audio.pacs.test_snk_pac_src_pac:
-    platform_allow:
-      - native_sim
-    integration_platforms:
-      - native_sim
     extra_configs:
       - CONFIG_BT_PAC_SNK=y
       - CONFIG_BT_PAC_SNK_LOC=n
       - CONFIG_BT_PAC_SRC=y
       - CONFIG_BT_PAC_SRC_LOC=n
   bluetooth.audio.pacs.test_src_pac_src_loc:
-    platform_allow:
-      - native_sim
-    integration_platforms:
-      - native_sim
     extra_configs:
       - CONFIG_BT_PAC_SNK=n
       - CONFIG_BT_PAC_SNK_LOC=n
       - CONFIG_BT_PAC_SRC=y
       - CONFIG_BT_PAC_SRC_LOC=y
   bluetooth.audio.pacs.test_src_pac:
-    platform_allow:
-      - native_sim
-    integration_platforms:
-      - native_sim
     extra_configs:
       - CONFIG_BT_PAC_SNK=n
       - CONFIG_BT_PAC_SNK_LOC=n

